vip-jetpack/connection-pilot/class-jetpack-connection-controls.php

```diff
@@ -168,31 +167,31 @@ public static function connect_site( bool $skip_connection_tests = false, bool $
 	 * Uses Akismet's function to connect Akismet using the Jetpack. An active Jetpack connection on the site
 	 * and the Akismet plugin are required.
 	 *
-	 * @return bool True if connection worked, false otherwise
+	 * @return mixed bool|WP_Error True if Akisment had a connection or was (re)connected, WP_Error otherwise.
 	 */
 	public static function connect_akismet(): bool {
-		if ( class_exists( 'Akismet_Admin' ) && method_exists( 'Akismet_Admin', 'connect_jetpack_user' ) ) {
-
-			if ( is_akismet_key_invalid() ) {
-				$original_user = wp_get_current_user();
-				// Getting wpcomvip user, since it's the owner of the Jetpack connection
-				$vip_user = get_user_by( 'login', 'wpcomvip' );
-				if ( ! $original_user || ! $vip_user ) {
-					return false;
-				}
+		if ( ! class_exists( 'Akismet_Admin' ) || ! method_exists( 'Akismet_Admin', 'connect_jetpack_user' ) || ! function_exists( 'is_akismet_key_invalid' ) ) {
+			return new WP_Error( 'jp-cxn-pilot-akismet-dependencies-missing', 'Akismet is missing required functions/methods to perform the connection.' );
```

Comment:
This doesn't satisfy `bool` return type.

Reply:
nice catch! fixed